src/cljs/rems/administration/catalogue_items.cljs

```diff
@@ -63,39 +65,31 @@
           :error-handler (flash-message/default-error-handler :top description)})
    {}))
 
-(rf/reg-event-db
- ::set-selected-items-ids
- (fn [db [_ items]]
-   (assoc db ::selected-items-ids items)))
-
 (rf/reg-sub
- ::selected-items-ids
- (fn [db _]
-   (::selected-items-ids db)))
-
-(defn- items-by-ids [items ids]
-  (filter (comp ids :id) items))
+ ::catalogue-by-ids
+ :<- [::catalogue]
+ (fn [items]
+   (index-by :id items)))
 
 (rf/reg-sub
  ::selected-catalogue-items
- (fn [_ _]
-   [(rf/subscribe [::catalogue])
-    (rf/subscribe [::selected-items-ids])])
- (fn [[catalogue selected-item-ids] _]
-   (items-by-ids catalogue selected-item-ids)))
+ :<- [::catalogue-by-ids]
+ :<- [:rems.table/selected-rows {:id ::catalogue}]
+ (fn [[catalogue-by-ids selected-item-ids] _]
+   (select-vals catalogue-by-ids selected-item-ids)))
 
 (defn- create-catalogue-item-button []
   [atoms/link {:class "btn btn-primary" :id :create-catalogue-item}
    "/administration/catalogue-items/create"
    (text :t.administration/create-catalogue-item)])
 
-(defn- change-form-button [items]
+(defn- update-catalogue-item-button [items]
   [:button.btn.btn-primary
    {:disabled (when (empty? items) :disabled)
     :on-click (fn []
-                (rf/dispatch [:rems.administration.change-catalogue-item-form/enter-page items])
-                (navigate! "/administration/catalogue-items/change-form"))}
-   (text :t.administration/change-form)])
+                (rf/dispatch [:rems.administration.update-catalogue-item/enter-page items])
+                (navigate! "/administration/catalogue-items/update-catalogue-item"))}
+   (text :t.administration/update-catalogue-item)])
```

Comment:
we could show count of the selected items here for visual confirmation, like so:

<img width="350" alt="Screenshot 2024-04-08 at 16 18 01" src="https://github.com/CSCfi/rems/assets/794087/86002d2f-b28e-406d-9af1-b0df500aead1">

what do you think?

Reply:
Maybe the button should also use a plural label. Maybe it could even be `Update catalogue items ... (3 selected)` or so?

I think it makes to spin this out as a further enhancement issue/PR.

Reply:
https://github.com/CSCfi/rems/issues/3280